src/azure-cli-testsdk/azure/cli/testsdk/preparers.py

```diff
@@ -22,6 +22,9 @@ def __init__(self, name_prefix='clitest.rg',
                  dev_setting_location='AZURE_CLI_TEST_DEV_RESOURCE_GROUP_LOCATION',
                  random_name_length=75):
         super(ResourceGroupPreparer, self).__init__(name_prefix, random_name_length)
+        if get_active_api_profile() == '2017-03-09-profile':
```

Comment:
Do not do this. The profile condition shouldn't invade into the basic components. Write a separate test case and preparer. The only case we have to add the condition is in storage account preparer, which is really a special case because of their breaking change.